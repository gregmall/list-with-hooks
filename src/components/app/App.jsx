import React from 'react';
import {
    BrowserRouter as Router,
    Route,
    Switch,
    
} from 'react-router-dom';
import List from '../../containers/list/List'
import Detail from '../../containers/detail/Detail'

export default function App() {

 return (
  <>
  <div className="app">

    <Router>
        <Switch>
             <Route exact path="/" component = {List} >
             </Route>
             <Route exact path="/detail/:id" component = {Detail}>
             </Route>
                
        </Switch>
    </Router>
  </div>
</>
  );
  
}
