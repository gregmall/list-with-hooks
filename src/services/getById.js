export const getById= (id) =>{
  return fetch(`https://rickandmortyapi.com/api/character/${id}`)
  .then(res => res.json());
  
};
