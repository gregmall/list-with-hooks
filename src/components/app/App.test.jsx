import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from './App';
import { getNameApi } from '../../services/getNameApi';

jest.mock('../../services/getNameApi.js');

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    getNameApi.mockResolvedValue([
      { name: 'morty', image: 'www.url.com' }, { name: 'mike', image: 'www.page.com' }]);
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
