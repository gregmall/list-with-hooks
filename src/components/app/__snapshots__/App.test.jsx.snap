// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <ul
      data-testid="characters"
    />
  </div>
</DocumentFragment>
`;
